lib/repo: For bare-user, make repo modes mirror checkout user-mode

cgwalters opened the pull request:
Having every object in a bare-user repo (and checkouts) be executable
is ugly.  I can't think of a good reason to do that.  So make
the committed mode semantics mirror that for user-mode checkouts; we
strip suid/sgid bits.

However, we also do ensure that the written files are read/writable by the
owning user, since otherwise we couldn't do anything to them. I'm not aware of
any real use cases for non-readable/non-writable by owner files in ostree.

Closes: https://github.com/ostreedev/ostree/issues/907

cgwalters commented:
Oh, interesting.  So this started failing in `bare-user-only` because the test suite does:

```
$OSTREE checkout test2-override checkout-test2-override
test -g checkout-test2-override/a/nested/2
```

For `bare-user`, because we weren't passing `-U` we ended up with a copy.  For `bare-user-only`, we suddenly started dropping suidness.

@alexlarsson - do you want ostree to preserve suid files in `bare-user-only`?  It seems like a bad idea, no?   Hmm, couldn't an app write a setuid file via the system helper today?

alexlarsson commented:
No that sounds pretty bad. We probably want to strip setuid in bare-user-only repos (i.e. lose fidelity for such objects).

cgwalters commented:
I wrote https://github.com/flatpak/flatpak/pull/837 - let's discuss this there.

cgwalters commented:
OK, updated this based on discussion in https://github.com/ostreedev/ostree/issues/907
Tests should pass now that we also fixed the suid bits in https://github.com/ostreedev/ostree/pull/909

cgwalters commented:
Argh, running into `umask` issues - at least in Fedora unprivileged users get `002`, but it looks like our Docker containers get `022.`  Added a fixup :arrow_up:

alexlarsson commented:
@rh-atomic-bot r+ 8fadd15
